File: MinecraftPvpPlugin/src/main/java/me/minecraft/minecraftpvpplugin/MinecraftPvpPlugin.java
package me.minecraft.minecraftpvpplugin;

import org.bukkit.*;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.block.Action;
import org.bukkit.event.entity.PlayerDeathEvent;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.player.*;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.plugin.java.JavaPlugin;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

import java.util.ArrayList;
import java.util.List;


public final class MinecraftPvpPlugin extends JavaPlugin implements Listener{

    static List<World> PVPWorlds = new ArrayList<World>();

    @Override
    public void onEnable() {
        // Plugin startup logic
        System.out.println("MAGIC PVP v.0.0.0");
        getServer().getPluginManager().registerEvents(this, this);
        World lobbyWorld = Bukkit.createWorld(new WorldCreator("Lobby"));

        for (int i = 0; i < 3; i++) {
            World world = Bukkit.createWorld(new WorldCreator(String.format("PVP%d", i)));
            world.setPVP(true);
            PVPWorlds.add(world);
        }

        if (lobbyWorld != null) {
            getLogger().info("Lobby world loaded successfully.");
        } else {
            getLogger().warning("Failed to load Lobby world.");
        }
        for (World world : Bukkit.getWorlds()) {
            getLogger().info("Loaded world: " + world.getName());
        }

        lobbyWorld.setPVP(false);
    }

    @EventHandler
    public void ClickEvent(InventoryClickEvent event){
        Player player = (Player) event.getWhoClicked();

        if (event.getClickedInventory() != null) {
            if (event.getClickedInventory().getTitle().equalsIgnoreCase(ChatColor.AQUA+"Join Game")){
                switch (event.getCurrentItem().getType()){
                    case DIAMOND_AXE:
                        PvpPlace.AddPlayer(player);
                        player.closeInventory();
                        break;
                }
                event.setCancelled(true);
            }
        }
    }

    @Override
    public boolean onCommand(CommandSender sender, Command command, String label, String[] args){
        if (command.getName().equalsIgnoreCase("lobby")) {
            Player player = (Player) sender;
            PvpPlace.RemovePlayer(player);
            ToLobby(player);
        }

        return true;
    }

    @EventHandler
    public void Join(PlayerJoinEvent event){
        Player player = event.getPlayer();
        event.setJoinMessage(ChatColor.AQUA + "Welcome " + player.getName());
        System.out.print(player.getName() + " join the server");
        player.teleport(Locations.lobby);
        ToLobby(player);
    }

    @EventHandler
    public void Click(PlayerInteractEvent event){
        Player player = event.getPlayer();
        if(player.getWorld()==Bukkit.getWorld("Lobby")){
            Inventory gui = Bukkit.createInventory(player, 9, ChatColor.AQUA+"Join Game");
            ItemStack StartGame = Items.DiamondPickaxe;
            ItemStack[] menu = {StartGame};
            gui.setContents(menu);

            if (event.getItem() != null) {
                if (event.getItem().equals(Items.DiamondSword)) {
                    if (event.getAction() != Action.PHYSICAL) {
                        player.openInventory(gui);
                    }
                }
            }
        }
        else{
            return;
        }
    }
    @EventHandler
    public void onPlayerDropItem(PlayerDropItemEvent event){
        Player player = event.getPlayer();
        if(Bukkit.getWorld("Lobby")==player.getWorld()){
            player.sendMessage("U cant drop any item ok?");
            event.getItemDrop().remove();
            player.getInventory().setItem(0, event.getItemDrop().getItemStack());
        }
    }

    @EventHandler
    public void PlayerDeathEvent(PlayerDeathEvent event) {
        Player player = event.getEntity();
        player.spigot().respawn();
        PvpPlace.RemovePlayer(player);
        event.setKeepInventory(true);
    }

    @EventHandler
    public void PlayerQuitEvent(PlayerQuitEvent event) {
        Player player = event.getPlayer();
        if (PVPWorlds.contains(player.getWorld())) {
            PvpPlace.RemovePlayer(player);
        }
    }

    static void ToLobby(Player player) {
        if (player != null) {
            System.out.println("To Lobby " + player.getName());
            player.getInventory().clear();
            Items.SwordItemMeta.setDisplayName(ChatColor.AQUA + "Join");
            Items.SwordItemMeta.spigot().setUnbreakable(true);
            Items.DiamondSword.setItemMeta(Items.SwordItemMeta);
            player.getInventory().setItem(0, Items.DiamondSword);
            player.getInventory().setHelmet(null);
            player.getInventory().setChestplate(null);
            player.getInventory().setLeggings(null);
            player.getInventory().setBoots(null);
            player.setHealth(20);
            player.setFoodLevel(20);
            ClearEffects(player);
            player.setGameMode(GameMode.ADVENTURE);
        }
    }

    static void ToPVP(Player player) {
        if (player != null) {
            player.getInventory().clear();
            player.getInventory().setItem(0, Items.IronSword);
            player.getInventory().setItem(1, Items.FishingRod);
            player.getInventory().setItem(8, Items.Gapple);
            player.setHealth(20);
            player.setFoodLevel(20);
            player.getInventory().setHelmet(Items.IronHelmet);
            player.getInventory().setChestplate(Items.IronChestplate);
            player.getInventory().setLeggings(Items.IronLeggings);
            player.getInventory().setBoots(Items.IronBoots);
            player.setGameMode(GameMode.ADVENTURE);
        }
    }

    static void ClearEffects (Player player){
        for (PotionEffect effect :player.getActivePotionEffects ()){
            player.removePotionEffect (effect.getType ());
        }
    }
}

File: MinecraftPvpPlugin/src/main/java/me/minecraft/minecraftpvpplugin/MinecraftPvpPlugin.kt
package me.minecraft.minecraftpvpplugin

import me.minecraft.minecraftpvpplugin.refs.Items
import me.minecraft.minecraftpvpplugin.refs.Locations
import me.minecraft.minecraftpvpplugin.refs.Skills
import me.minecraft.minecraftpvpplugin.refs.Worlds
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.GameMode
import org.bukkit.Material
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.block.Action
import org.bukkit.event.entity.PlayerDeathEvent
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.player.PlayerDropItemEvent
import org.bukkit.event.player.PlayerInteractEvent
import org.bukkit.event.player.PlayerJoinEvent
import org.bukkit.event.player.PlayerQuitEvent
import org.bukkit.inventory.ItemStack
import org.bukkit.plugin.java.JavaPlugin
import java.util.logging.Level

class MinecraftPvpPlugin : JavaPlugin(), Listener {

    override fun onEnable() {
        logger.level = Level.ALL

        // Plugin startup logic
        logger.info("SKYJET PVP v.3.0.0")
        server.pluginManager.registerEvents(this, this)

        Worlds.lobby.pvp = false

        Worlds.lobby.setGameRuleValue("doMobSpawning", "false")
        Worlds.lobby.setGameRuleValue("doDaylightCycle", "true")
        Worlds.lobby.setGameRuleValue("doWeatherCycle", "true")

        Worlds.pvpWorlds.map {
            it.pvp = true
            it.setGameRuleValue("doMobSpawning", "false")
            it.setGameRuleValue("doDaylightCycle", "false")
            it.setGameRuleValue("doWeatherCycle", "false")
            it.setGameRuleValue("mobGriefing", "false")
            it.setGameRuleValue("fireTick", "false")
        }



        instance = this
    }

    override fun onCommand(sender: CommandSender, command: Command, label: String, args: Array<String>): Boolean {
        if (command.name.equals("lobby", ignoreCase = true)) {
            val player = sender as Player
            PvpPlaceManager.removePlayer(player)
            onPlayerToLobby(player)
        }

        return true
    }

    @EventHandler
    fun handleInventoryClick(event: InventoryClickEvent) {
        val player = event.whoClicked as Player

        if (event.clickedInventory == null) return

        if (event.clickedInventory.title.equals("${ChatColor.AQUA}Join Game", ignoreCase = true)) {
            when (event.currentItem.type) {
                Material.DIAMOND_AXE -> {
                    val selectGui = Bukkit.createInventory(player, 9, "${ChatColor.AQUA}Select skill")
                    selectGui.contents = Skills.skills
                    player.openInventory(selectGui)
                }

                else -> {}
            }

            event.isCancelled = true
        }

        if (event.clickedInventory.title.equals("${ChatColor.AQUA}Select skill", ignoreCase = true)) {
            Skills.skills.forEach {
                if (it.name == event.currentItem.itemMeta.displayName) {
                    PvpPlaceManager.addPlayer(player, it)
                }
            }

            event.isCancelled = true
        }
    }

    @EventHandler
    fun handlePlayerJoin(event: PlayerJoinEvent) {
        val player = event.player

        event.joinMessage = "${ChatColor.AQUA}Welcome ${player.name}!"
        logger.info("${player.name} joined the server")
        player.teleport(Locations.lobbySpawn)
        onPlayerToLobby(player)
    }

    @EventHandler
    fun handlePlayerInteract(event: PlayerInteractEvent) {
        val player = event.player

        if (player.world == Worlds.lobby) {
            if (event.item == null) return
            if (event.item == Items.diamondSword && event.action != Action.PHYSICAL) {
                val startGui = Bukkit.createInventory(player, 9, "${ChatColor.AQUA}Join Game")
                val joinGameBtn = Items.diamondPickaxe

                startGui.contents = arrayOf(joinGameBtn)

                player.openInventory(startGui)
            }
        }
    }

    @EventHandler
    fun handlePlayerDropItem(event: PlayerDropItemEvent) {
        val player = event.player

        if (player.world == Worlds.lobby) {
            player.sendMessage("U cant drop any item ok?")
            event.itemDrop.remove()
            player.inventory.setItem(0, event.itemDrop.itemStack)
        }
    }

    @EventHandler
    fun handlePlayerDeath(event: PlayerDeathEvent) {
        val player = event.entity

        player.spigot().respawn()
        PvpPlaceManager.removePlayer(player)
        event.keepInventory = true
    }

    @EventHandler
    fun handlePlayerQuit(event: PlayerQuitEvent) {
        val player = event.player

        if (Worlds.isInPvp(player)) {
            PvpPlaceManager.removePlayer(player)
        }
    }

    companion object {
        var instance: JavaPlugin? = null

        fun onPlayerToLobby(player: Player) {
            println("To Lobby " + player.name)
            player.inventory.clear()
            Items.swordItemMeta.displayName = "${ChatColor.AQUA}Join"
            Items.swordItemMeta.spigot().isUnbreakable = true
            Items.diamondSword.setItemMeta(Items.swordItemMeta)
            player.inventory.setItem(0, Items.diamondSword)
            player.inventory.helmet = null
            player.inventory.chestplate = null
            player.inventory.leggings = null
            player.inventory.boots = null
            player.health = 20.0
            player.foodLevel = 20
            clearEffects(player)
            player.gameMode = GameMode.ADVENTURE
        }

        fun onPlayerToPvp(player: Player, skillItem: ItemStack) {
            player.inventory.clear()
            player.inventory.setItem(0, Items.ironSword)
            player.inventory.setItem(1, Items.fishingRod)
            player.inventory.setItem(2, skillItem)

            player.inventory.setItem(8, Items.gapple)
            player.health = 20.0
            player.foodLevel = 20
            clearEffects(player)
            player.inventory.helmet = Items.ironHelmet
            player.inventory.chestplate = Items.ironChestplate
            player.inventory.leggings = Items.ironLeggings
            player.inventory.boots = Items.ironBoots
            player.gameMode = GameMode.ADVENTURE
        }

        private fun clearEffects(player: Player) {
            for (effect in player.activePotionEffects) {
                player.removePotionEffect(effect.type)
            }
        }
    }
}